prediction_market_agent/agents/crewai_subsequential_agent/crewai_agent_subquestions.py

```diff
@@ -0,0 +1,140 @@
+                    verbose=True,
+                    allow_delegation=False,
+                    tools=[]
+                )
+
+    def split_research_into_outcomes(self, question: str) -> Outcomes:
+        create_outcomes_task = Task(
+          description=(CREATE_OUTCOMES_FROM_SCENARIO_PROMPT),
+          expected_output=CREATE_OUTCOMES_FROM_SCENARIO_OUTPUT,
+          tools = [],
+          output_json=Outcomes,
+          agent=self.researcher,
+        )
+
+        report_crew = Crew(
+            agents=[self.researcher],
+            tasks=[create_outcomes_task],
+        )
+        result = report_crew.kickoff(inputs={'scenario': question})
+        return result
+
+    def generate_prediction_for_one_outcome(self, sentence: str) -> ProbabilityOutput:
+
+        task_research_one_outcome = Task(
+                    description=(RESEARCH_OUTCOME_PROMPT),
+                    tools=[search_tool],
+                    agent=self.researcher,
+                    expected_output=(RESEARCH_OUTCOME_OUTPUT),
+                )
+        task_create_probability_for_one_outcome = Task(
+                    description=(PROBABILITY_FOR_ONE_OUTCOME_PROMPT),
+                expected_output=PROBABILITY_CLASS_OUTPUT,
+                    agent=self.predictor,
+                output_json=ProbabilityOutput,
+            context=[task_research_one_outcome]
+                )
+        crew = Crew(
+                    agents=[self.researcher, self.predictor],
+                    tasks=[task_research_one_outcome, task_create_probability_for_one_outcome],
+                    verbose=2,
+                process=Process.sequential            
+                )
+        
+        result = crew.kickoff(inputs={'sentence': sentence})
+        return result
+
+    def generate_final_decision(self, outcomes_with_probabilities) -> ProbabilityOutput:
+
+        task_final_decision = Task(
+                    description=(FINAL_DECISION_PROMPT),
+                    tools=[],
+                    agent=self.predictor,
+                    expected_output=(PROBABILITY_CLASS_OUTPUT),
+                output_json=ProbabilityOutput,
+                )
+        
+        crew = Crew(
+                    agents=[self.predictor],
+                    tasks=[task_final_decision],
+                    verbose=2, 
+                )
+        
+        result = crew.kickoff(inputs={'outcomes_with_probabilities': 
+                                    [(i[0],i[1].dict()) for i in outcomes_with_probabilities],
+                                    'number_of_outcomes': len(outcomes_with_probabilities),
+                                    'outcome_to_assess': outcomes_with_probabilities[0][0]})
+        return result
+
+    def answer_binary_market(self, market: AgentMarket) -> bool:
+        
+        outcomes = self.split_research_into_outcomes(market.question)
+
+        outcomes_with_probs = []
+        for outcome in tqdm(outcomes.outcomes):
```

Comment:
So what you're doing here is independently getting predictions for each outcome, which together make up the full 'probability space', to make sure the predictions are consistent. That sounds like a good idea.

I feel like it's going to be quite hard to evaluate the effect of this feature alone, as this multi-agent also differs significantly from the evo agent in that it has this crewai-based tool use. Wonder if it makes sense to try adding this feature to the evo agent first to get a sense of whether it's a good idea.

Reply:
I created a benchmark for this, but agree that it will be hard to benchmark this due to crewai usage.
I suggest we simply compare the precisions on Dune or sample from the answers on a few resolved markets and calculate metrics for that, what do you think?